libvmi/read.c

```diff
@@ -97,13 +97,13 @@ vmi_read(
 
         if(dtb) {
             paddr = vmi_pagetable_lookup(vmi, dtb, start_addr + buf_offset);
+            if (!paddr) {
```

Comment:
Probably a better approach would be to have vmi_pagetable_lookup return an errno in case something went wrong during the translation, instead of relying on a potentially valid paddr=0 being an error condition.